File: c/fib.cxx
#include <iostream>
using lint = unsigned long long int;
using namespace std;

lint a(0);
lint b(1);
lint c;

inline void put(lint i) {
  cout << i << ' ';
}

void print_usage() {
  cerr << "fib n\n  n: the number of numbers you want to generate\n";
  exit(1);
}

int main(int argc, char *argv[]) {
  if (argc != 2) {
    print_usage();
  }
  int count;
  try {
    count = stoi(argv[1]);
  } catch (exception e) {
    print_usage();
  }
  if (count > 0) {
    put(0);
    if (count > 1) {
      put(1);
      count -= 2;
      while (count --> 0) {
        c = a + b;
        a = b;
        b = c;
        put(c);
      }
    }
  }
  cout << '\n';
  return 0;
}

File: js/fib.js
#!/usr/bin/env node

var a = 0|0;
var b = 1|0;
var c = 0|0;
var count = process.argv[2]|0;
function put(i) {
   i |= 0;
   process.stdout.write((i|0) + " ");
}
if ((count|0) > 0) {
   put(0);
   if ((count|0) > 1) {
      put(1);
      if ((count|0) > 2) {
         count = (count|0) - 2;
         while (count > 0) {
            c = (a|0) + (b|0);
            a = (b|0);
            b = (c|0);
            put(c|0);
            count = (count|0) - 1;
         }
      }
   }
}
process.stdout.write('\n');
process.exit(0);
